### PR TITLE
add ssh port to inventory

### PR DESCRIPTION
**Pull request by dangel101**
Bug-Url: https://bugzilla.redhat.com/show_bug.cgi?id=2104939

### CODE REVIEW

**Review by mwperina — APPROVED** [review]
+1

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleCommandConfig.java [hunk 1]
```diff
@@ -307,14 +301,16 @@ private File setupAnsibleRunnerExecutionDir() throws IOException {
     }
 
     private void createHostFile(File inventory) {
-        File hostsFile = new File(String.format("%1$s/hosts", inventory));
-        if (CollectionUtils.isNotEmpty(hostnames())) {
+        File hostFile = new File(String.format("%1$s/hosts", inventory));
+        if (!hostname.isEmpty()) {
```
Comment (smelamud) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleCommandConfig.java:305]: Can it be null? `!StringUtils.isEmpty(hostname)` may be better.

**Review by smelamud — CHANGES_REQUESTED** [review]

**Review by ahadas — COMMENTED** [review]
besides the comment inside on keep checking for null value of `hostname`, this looks great, thanks @dangel101

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleCommandConfig.java (hunk 1 above)
Comment (dangel101) [reply backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleCommandConfig.java]: I removed this property completely now

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleCommandConfig.java [hunk 2]
```diff
@@ -307,15 +295,15 @@ private File setupAnsibleRunnerExecutionDir() throws IOException {
...
-                Files.write(hostsFile.toPath(), StringUtils.join(hostnames(), System.lineSeparator()).getBytes());
-            } catch (IOException ex) {
-                throw new AnsibleRunnerCallException(
-                    String.format("Failed to create inventory file '%s':", hostsFile.toString()), ex);
-            }
+        File hostFile = new File(String.format("%1$s/hosts", inventory));
+        try {
+            hostFile.createNewFile();
+            Files.write(hostFile.toPath(),
+                    String.format(this.host.getHostName() + " ansible_port=%1$s", this.host.getSshPort()).getBytes());
```
Comment (smelamud) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleCommandConfig.java:302]: Can `this.host` be null? I think so, looking to `AnsibleExecutor.createAuditLogable()` below.

**Review by smelamud — CHANGES_REQUESTED** [review]

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleExecutor.java [hunk 3]
```diff
@@ -190,14 +190,14 @@ public AnsibleReturnValue runCommand(AnsibleCommandConfig commandConfig, int tim
     }
 
     private AuditLogable createAuditLogable(AnsibleCommandConfig command, String taskName) {
-        if (command.hosts() == null) {
+        if (command.host() == null) {
```
Comment (mwperina) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleExecutor.java:193]: We can remove the check if we add a check to AnsibleExecutor.runCommand

**Review by mwperina — CHANGES_REQUESTED** [review]

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleExecutor.java (hunk 3 above)
Comment (dangel101) [reply backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleExecutor.java]: done

**Review by smelamud — APPROVED** [review]

**Comment by smelamud** [comment]
Please add a description to the commit message.

**Review by ahadas — APPROVED** [review]

**Review by mwperina — APPROVED** [review]
+1

**Pull request merged**